core/p2p/p2pv1/server.go

```diff
@@ -108,8 +114,26 @@ func (p *P2PServerV1) Init(cfg config.P2PConfig, lg log.Logger, extra map[string
 			p.staticNodes["xuper"] = peerids
 		}
 	}
-
+	p.connectToBootNodes()
+	p.registerSubscribe()
 	go p.Start()
+	go p.handleMsg()
+	return nil
+}
+
+// ConnectToBootNodes connect to bootnode
+func (p *P2PServerV1) connectToBootNodes() error {
+	msg, err := p2p_base.NewXuperMessage(p2p_base.XuperMsgVersion3, "", "", p2pPb.XuperMessage_NEW_NODE, []byte{}, p2pPb.XuperMessage_NONE)
+	if err != nil {
+		p.log.Error("ConnectToBootNodes NewXuperMessage error", "error", err)
+		return err
+	}
+	msg.Header.From = strconv.Itoa(int(p.config.Port))
+	p.ConnectToPeersByAddr(p.bootNodes)
```

Comment:
Looks like no peer exchange message?  How can this node get other nodes from bootstrap node?

Reply:
A file `msg_processor.go` was missed in last commit. Now p2pv1 is mainly used static nodes as routes. But when new node joins, one nodes needs to restart to update the config. So this commit support nodes joined with not restart any nodes of the network. 
While a nodes joined, bootNodes will add the nodes to it `dynamic nodes` and the default broadcast strategy contain `static nodes`, `boot nodes` and `dynamic nodes`.